tests/trees/test_isoup_tree.py

```diff
@@ -123,7 +125,8 @@ def test_isoup_tree_adaptive(test_path):
 
     assert np.allclose(y_pred, expected_predictions)
     error = mean_absolute_error(y_true, y_pred)
-    expected_error = 150.63043466390528
+
+    expected_error = 152.8716829154756
```

Comment:
Same comment about clarifying changes in model performance.

Reply:
Once again, the performance have changed since the nodes are only updating the weights of their own perceptrons, rather than changing a shared region in memory. My hypothesis is that in all the cases the performance has decreased due to the characteristics of the synthetic dataset (linear relationships between inputs and target(s)).